protosocket-connection/src/connection.rs

```diff
@@ -56,8 +58,26 @@ impl<Bindings: ConnectionBindings> Unpin for Connection<Bindings> {}
 impl<Bindings: ConnectionBindings> Future for Connection<Bindings> {
     type Output = ();
 
+    /**
+     * The poll method is the core of the Connection's asynchronous behavior.
+     * It's recommended to familiarize yourself with the ConnectionBindings trait
+     * in the types.rs file of this crate before proceeding.
+     *
+     * This method performs the following steps:
+     *
+     * 1. Check for read readiness on the Connection's TcpStream.
+     * 2. If data is available, read raw bytes into the receive_buffer (up to max_buffer_length).
+     * 3. Deserialize the read bytes into Messages and store them in the inbound_messages queue.
+     * 4. Process all messages in the inbound queue using the user-provided MessageReactor.
+     * 5. Check the outbound_messages queue for messages to send.
+     * 6. If outbound messages exist, check for write readiness and send serialized messages.
+     *
+     * The method returns Poll::Ready(()) when the connection is closed (gracefully or due to an error).
+     * It returns Poll::Pending when the connection is still active and may have more work to do in the future.
+     */
     fn poll(mut self: Pin<&mut Self>, context: &mut Context<'_>) -> Poll<Self::Output> {
```

Comment:
This is a rust repo - let's use rust comments rather than javadocs: `///` is the multiline comment prefix for rust symbols that you want to have pop up in IDEs. See other commented symbols in this file for examples of the idiomatic way to write multiline rust symbol comments.